squidDrone/src/com/com_message_buffer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef SRC_COM_MESSAGE_BUFFER_HPP_
+#define SRC_COM_MESSAGE_BUFFER_HPP_
+
+#include <array>
+#include <queue>
+#include "com_types.hpp"
+
+namespace com {
+  /** 
+    * @brief This class represents a queue style buffer for communication
+    * frames exchanged between drones.
+    * 
+    */
+  class ComMessageBuffer {
+    public:
+      /** 
+       * @brief Construct a new Com Message Buffer object
+       * 
+       */
+      ComMessageBuffer(){};
+
+      /** 
+       * @brief Transfer data to queue buffer. 
+       * 
+       * @param data Reference to data frame. Must conform to data frame specifications.
+       * @return uint8_t Returns 0 on no error, otherwise negative value.
+       * 
+       */
+      types::ComError put_data(std::array<std::uint8_t, 32> &data);
```

Comment:
No negative enum value for ComError is defined. The current enum class ComError can never be below zero. Either change @return documentation or add a enum member with negativ value and return it explicitly

Reply:
Forgot to update the doc. Fixed